observatory_platform/templates/updated_example/specialised_telescope.py

```diff
@@ -0,0 +1,130 @@
+from abc import ABC, abstractmethod
+from airflow import DAG
+from airflow.operators.python_operator import PythonOperator, ShortCircuitOperator
+from airflow.models.taskinstance import TaskInstance
+import pendulum
+from observatory_platform.templates.updated_example.telescope import TelescopeRelease, Telescope
+import logging
+from datetime import timedelta
+
+
+class SpecialisedRelease(TelescopeRelease):
+    """ Used to store info on a given release """
+    def __init__(self, dag_id: str, start_date: pendulum.Pendulum, end_date: pendulum.Pendulum,
+                 extensions: dict, first_release: bool = False):
+        super().__init__(dag_id, start_date, end_date, extensions, first_release)
+
+
+def pull_release(ti: TaskInstance) -> SpecialisedRelease:
+    """ Pulls xcom with release.
+
+    :param ti: Task instance
+    :return: One release
+    """
+    return ti.xcom_pull(key=SpecialisedTelescope.RELEASES_XCOM, task_ids=SpecialisedTelescope.TASK_ID_CREATE_RELEASE,
+                        include_prior_dates=False)
+
+
+def pull_last_start_date(ti: TaskInstance) -> pendulum.Pendulum:
+    """ Pulls xcom with last start date of this run.
+
+    :param ti: Task instance
+    :return: Last start date
+    """
+    return ti.xcom_pull(key=SpecialisedTelescope.LAST_START_XCOM, task_ids=SpecialisedTelescope.TASK_ID_CREATE_RELEASE,
+                        include_prior_dates=True)
+
+
+class SpecialisedTelescope(Telescope, ABC):
+    RELEASES_XCOM = "release"
+    LAST_START_XCOM = "last_start"
+
+    TASK_ID_CREATE_RELEASE = "create_release"
+    TASK_ID_DOWNLOAD = "download"
+    TASK_ID_BQ_LOAD_PARTITION = "bq_load_partition"
+
+    def __init__(self, dag_id, start_date, schedule_interval, extensions, partition_table_id):
+        super().__init__(dag_id, start_date, schedule_interval, extensions)
+        self.partition_table_id = partition_table_id
+
+    def create_airflow_DAG(self) -> DAG:
```

Comment:
The naming was on purpose, so that 'airflow' and 'DAG' are both mentioned whenever you create the DAG, to avoid this issue with airflow not finding the dag. (https://airflow.apache.org/docs/stable/faq.html) Maybe there's a better workaround, especially in the 'new' set-up with split repositories.